consul/consul.go

```diff
@@ -1,97 +1,52 @@
 package consul
 
 import (
+	"github.com/alaa/pencil-go/registry"
 	consul "github.com/hashicorp/consul/api"
 )
 
-type ConsulClient struct {
-	client *consul.Client
+// ServiceRepository is consul-based implementation of registry.ServiceRepository
+type ServiceRepository struct {
+	consulAgent consulAgent
 }
 
-func NewConsulClient() (ConsulClient, error) {
-	client, err := consul.NewClient(consul.DefaultConfig())
-	if err != nil {
-		return ConsulClient{}, err
-	}
-	return ConsulClient{client: client}, nil
-}
-
-type ConsulAgent struct {
-	agent *consul.Agent
+type consulAgent interface {
+	Services() (map[string]*consul.AgentService, error)
+	ServiceRegister(service *consul.AgentServiceRegistration) error
+	ServiceDeregister(serviceID string) error
 }
 
-func (c *ConsulClient) NewConsulAgent() ConsulAgent {
-	agent := c.client.Agent()
-	return ConsulAgent{agent: agent}
-}
-
-type Member struct {
-	Name string
-	IP   string
-	Port uint16
-}
-
-type Members []Member
-
-func buildMember(name string, ip string, port uint16) Member {
-	return Member{Name: name, IP: ip, Port: port}
-}
-
-func (a *ConsulAgent) members() Members {
-	list := Members{}
-	use_wan := false
-	members, _ := a.agent.Members(use_wan)
-	for _, member := range members {
-		list = append(list, buildMember(member.Name, member.Addr, member.Port))
-	}
-	return list
+// NewServiceRepository creates new instance of ServiceRepository structure
+func NewServiceRepository(consulAgent consulAgent) *ServiceRepository {
+	return &ServiceRepository{consulAgent}
 }
 
-func buildService(id string, name string, port int, ip string) consul.AgentServiceRegistration {
-	return consul.AgentServiceRegistration{ID: id, Name: name, Port: port, Address: ip}
+// Register adds service into consul
+func (c *ServiceRepository) Register(service *registry.Service) error {
```

Comment:
detail, but I don't understand why you've picked `c` as a receiver name here? wouldn't `r` or `repository` be a better fit? :)